lib/srv/discovery/status.go

```diff
@@ -60,7 +63,7 @@ func (s *Server) updateDiscoveryConfigStatus(discoveryConfigNames ...string) {
 		// Static configurations (ie those in `teleport.yaml/discovery_config.<cloud>.matchers`) do not have a DiscoveryConfig resource.
 		// Those are discarded because there's no Status to update.
 		if discoveryConfigName == "" {
-			return
+			continue
```

Comment:
This is a bugfix, right?

Reply:
Yes: mixing static and dynamic discovery configs may end up with loss of update for the dynamic ones due to this invalid `return`.

I can split it into separate PR if you wish?